### ISSUE TITLE
Default to styling all supported file types in `style_dir()` and `style_pkg()`?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/styler#963]
I am wondering if `style_dir()` and `style_pkg()` shouldn't style all supported file formats in the directory/package by default. 

Is there any reason why the users may wish to format only code in `.R` files, but not in (e.g.) `.Rmd` files? 

Current default:

```r
filetype = c("R", "Rprofile")
```

Proposed default:

```r
filetype = c("R", "Rprofile", "Rmd", "Rmarkdown", "Rnw")
```

**Comment by lorenzwalthert** [comment]
Yes. Also thought about it. Did not want to do that in the very beginning though, when formatting of non-R was experimental.

**Comment by lorenzwalthert** [comment]
This issue is also included in #769, although questionable if breaking or not. PR welcome yes.
